golang/src/main/java/com/regnosys/rosetta/generator/golang/enums/GolangEnumGenerator.xtend

```diff
@@ -0,0 +1,160 @@
+package com.regnosys.rosetta.generator.golang.enums
+
+import com.google.inject.Inject
+import com.regnosys.rosetta.generator.golang.object.GolangModelObjectBoilerPlate
+import com.regnosys.rosetta.rosetta.RosettaEnumValue
+import com.regnosys.rosetta.rosetta.RosettaEnumeration
+import java.util.ArrayList
+import java.util.HashMap
+import java.util.List
+import java.util.Map
+
+import static com.regnosys.rosetta.generator.golang.util.GolangModelGeneratorUtil.*
+import com.regnosys.rosetta.generator.java.enums.EnumHelper
+import java.nio.file.Files
+import java.nio.file.Paths
+
+class GolangEnumGenerator {
+	
+	@Inject extension GolangModelObjectBoilerPlate
+	
+	static final String FILENAME = 'enums.go'
+		
+	def Map<String, ? extends CharSequence> generate(Iterable<RosettaEnumeration> rosettaEnums, String version) {
+		val result = new HashMap
+		val enums = rosettaEnums.sortBy[name].generateEnums(version).replaceTabsWithSpaces
+		result.put(FILENAME,enums)
+		return result;
+	}
+	
+	//generates struct representations of enums
+	def Map<String, ? extends CharSequence> generate2(Iterable<RosettaEnumeration> rosettaEnums, String version) {
+		val result = new HashMap
+		val enums = rosettaEnums.sortBy[name].generateEnums2(version)
+		result.put(FILENAME,enums)
+		return result;
+	}
+	
+	def Map<String, ? extends CharSequence>generate3(Iterable<RosettaEnumeration> rosettaEnums, String version) {
+		val result = new HashMap
+		val enums = rosettaEnums.sortBy[name].generateEnums3(version)
+		result.put(FILENAME,enums)
+		return result;
+		
+	}
+
+	def static toJavaEnumName(RosettaEnumeration enumeration, RosettaEnumValue rosettaEnumValue) {
+		return enumeration.name + '.' + EnumHelper.convertValues(rosettaEnumValue)
+	}
+
+	private def allEnumsValues(RosettaEnumeration enumeration) {
+		val enumValues = new ArrayList
+		var e = enumeration;
+
+		while (e !== null) {
+			e.enumValues.forEach[enumValues.add(it)]
+			e = e.superType
+		}
+		return enumValues.sortBy[name];
+	}
+
+	private def generateEnums(List<RosettaEnumeration> enums, String version)  '''		
+		package enums
```

Comment:
package should be `org.isda.cdm`

Reply:
Golang does not allow dots in package names.